### PR TITLE
[AN-4532] Ephemeral message stuff

### PR DESCRIPTION
**Pull request by hakonbo**
#### APK
[Download build #7693](http://192.168.10.18:8080/job/Pull%20Request%20Builder/7693/artifact/build/artifact/wire-dev-PR224-7693.apk)
[Download build #7694](http://192.168.10.18:8080/job/Pull%20Request%20Builder/7694/artifact/build/artifact/wire-dev-PR224-7694.apk)
[Download build #7695](http://192.168.10.18:8080/job/Pull%20Request%20Builder/7695/artifact/build/artifact/wire-dev-PR224-7695.apk)

### CODE REVIEW

**Comment by hakonbo** [comment]
build again

**Comment by shumengye** [comment]
build again

File: wire-ui/src/main/java/com/waz/zclient/ui/views/EphemeralDotAnimationView.java [hunk 1]
```diff
@@ -0,0 +1,189 @@
...
+    private int selectedCount;
+    private Paint paint;
+    private ValueAnimator valueAnimator;
+
+    private ModelObserver<Message> messageModelObserver = new ModelObserver<Message>() {
+        @Override
+        public void updated(Message message) {
+            if (message.isEphemeral()) {
+                if (message.getExpirationTime() == Instant.MAX) {
+                    setVisibility(View.VISIBLE);
```
Comment (shumengye) [inline wire-ui/src/main/java/com/waz/zclient/ui/views/EphemeralDotAnimationView.java]: You can always set `setVisibility(View.VISIBLE)` when message is ephemeral and remove from here and `startAnimation`

File: wire-ui/src/main/java/com/waz/zclient/ui/views/EphemeralDotAnimationView.java [hunk 2]
```diff
@@ -0,0 +1,189 @@
...
+                    showAllDots();
+                } else {
+                    long diff = ChronoUnit.MILLIS.between(Instant.now(), message.getExpirationTime());
+                    if (diff > 0) {
+                        startAnimation(message.getEphemeralExpiration().milliseconds,
+                                                                 diff);
```
Comment (shumengye) [inline wire-ui/src/main/java/com/waz/zclient/ui/views/EphemeralDotAnimationView.java]: Indentation

File: wire-ui/src/main/java/com/waz/zclient/ui/views/EphemeralDotAnimationView.java [hunk 3]
```diff
@@ -0,0 +1,189 @@
...
+    private int secondaryColor;
+    private int selectedCount;
+    private Paint paint;
+    private ValueAnimator valueAnimator;
+
+    private ModelObserver<Message> messageModelObserver = new ModelObserver<Message>() {
+        @Override
+        public void updated(Message message) {
+            if (message.isEphemeral()) {
+                if (message.getExpirationTime() == Instant.MAX) {
```
Comment (shumengye) [inline wire-ui/src/main/java/com/waz/zclient/ui/views/EphemeralDotAnimationView.java]: Which case is that? ephemeral message doesn't expire?

**Comment by shumengye** [comment]
review done

**Comment by hakonbo** [comment]
build again

**Pull request merged**